src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunLibraryRootProvider.cs

```diff
@@ -13,30 +13,55 @@ namespace ILCompiler
     public class ReadyToRunRootProvider : ICompilationRootProvider
     {
         private EcmaModule _module;
+        private ProfileData _profileData;
 
-        public ReadyToRunRootProvider(EcmaModule module)
+        public ReadyToRunRootProvider(EcmaModule module, ProfileDataManager profileDataManager)
         {
             _module = module;
+            _profileData = profileDataManager.GetDataForModuleDesc(module);
```

Comment:
Should the constructor just take the `ProfileData` instead of `ProfileDataManager`?

Reply:
No, we may want in the future to pull profile data from multiple binaries.